streamingccl: enable lag replanning by default

msbutler opened the pull request:
See individual commits.

msbutler committed:
roachtest: add c2c/import/kv0 roachtests

This patch adds a c2c roachtest which runs a kv import after the replication
stream has started, inducing catchup scans during the kv import. This patch
adds a nightly test running a 1TB import and a weekly which runs a 7tb import.

Informs #122846

Release note: none

msbutler committed:
roachtest: run c2c/BulkOps not c2c/BulkOps/short

Epic: none

Release note: none

msbutler committed:
streamingccl: enable lag replanning by default

This patch enables automatic c2c distSQL replanning if one node is more than 5
minutes behind the mean node. This patch also introduces the
`stream_replication.lag_check_frequency` cluster setting, as the lag checking
frequency was previously set by `stream_replicaiton.replan_flow_frequency/2`,
which is extremely hard to reason about. Now the lag_check_frequency checks for
lagging nodes every 3 minutes by default and if one node is more than 5 minutes
behind, we replan.

Epic: none

Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/125789)
<!-- Reviewable:end -->

msbutler committed:
streamingccl: skip lag replanning during initial scan

This patch prevents replanning thrashing when a subset of nodes have completed
their initial scan.

Epic: none

Release note: none

msbutler commented:
I plan to backport this to 24.1 and call it for PCR bulk imports.

msbutler commented:
unrelated flake

msbutler commented:
TFTR! 

bors r=dt
